subsys/fs/nvs/nvs.c

```diff
@@ -781,10 +781,21 @@ static int nvs_startup(struct nvs_fs *fs)
 			}
 		}
 	}
-	/* all sectors are closed, this is not a nvs fs */
+	/* all sectors are closed, this is not a nvs fs or irreparably corrupted */
 	if (closed_sectors == fs->sector_count) {
+#ifdef CONFIG_NVS_BAD_PARTITION_RECOVERY
+		LOG_WRN("All sectors closed, erasing all settings...");
```

Comment:
There should be no mention of settings here.
NVS and Settings are two different subsystems.
You should use "erase all entries" or maybe "erasing all sectors"